src/main/java/com/dreammaster/gthandler/multiAirFilter/GT_MetaTileEntity_AirFilterBase.java

```diff
@@ -3,95 +3,135 @@
-        .addInfo("Add " + ItemList.AdsorptionFilter.getIS().getDisplayName() + " in input bus to double efficiency")
-		.addInfo("Machine tier = Maximum effective Muffler tier")
-		.addInfo("Features Hysteresis control (tm)")
-		.addInfo("Each muffler reduce pollution by 30 * TurbineEfficiency * Floor(2.5^Tier) every second")
-		.addSeparator()
-		.beginStructureBlock(3, 4, 3, true)
-		.addController("Front bottom")
-		.addOtherStructurePart("Air Filter Turbine Casing", "Top and bottom")
-		.addOtherStructurePart("Air Filter Vent Casing", "Corners of the middle 2 layers")
-		.addOtherStructurePart("Muffler Hatch Or Air Filter Vent Casing", "8 in the middle layers")
-		.addEnergyHatch("Any bottom layer casing")
-		.addMaintenanceHatch("Any bottom layer casing")
-		.addInputBus("Any bottom layer casing")		
-		.addOutputBus("Any bottom layer casing")		
-		.toolTipFinisher("GTNH Coremod");
-		if (!Keyboard.isKeyDown(Keyboard.KEY_LSHIFT)) {
-			return tt.getInformation();
-		} else {
-			return tt.getStructureInformation();
-		}  	
+        final GT_Multiblock_Tooltip_Builder tt = new GT_Multiblock_Tooltip_Builder();
+        tt.addMachineType("Air Filter")
+                .addInfo("Controller block for the Electric Air Filter T"+ multiTier)
+                .addInfo("Needs a Turbine in the controller")
+                .addInfo("Add " + ItemList.AdsorptionFilter.getIS().getDisplayName() + " in input bus to double efficiency (30 uses per item)")
+                .addInfo("Machine tier = Maximum effective Muffler tier")
+                .addInfo("Can process "+(2*multiTier+1)+"x"+(2*multiTier+1)+" chunks")
+                .addInfo("Each muffler reduce pollution by 30 * TurbineEfficiency * Floor(2.5^Tier) every second")
+                .addSeparator()
+                .beginStructureBlock(3, 4, 3, true)
+                .addController("Front bottom")
+                .addOtherStructurePart(GT_LanguageManager.getTranslation(GT_Container_CasingsNH.sBlockCasingsNH.getUnlocalizedName()+"."+getCasingMeta()+".name"), "Top and bottom")
+                .addOtherStructurePart(GT_LanguageManager.getTranslation(GT_Container_CasingsNH.sBlockCasingsNH.getUnlocalizedName()+"."+getPipeMeta()+".name"), "Corners of the middle 2 layers")
+                .addOtherStructurePart("Muffler Hatch Or "+GT_LanguageManager.getTranslation(GT_Container_CasingsNH.sBlockCasingsNH.getUnlocalizedName()+"."+getPipeMeta()+".name"), "8 in the middle layers")
+                .addEnergyHatch("Any bottom layer casing")
+                .addMaintenanceHatch("Any bottom layer casing")
+                .addInputBus("Any bottom layer casing")
+                .addOutputBus("Any bottom layer casing")
+                .toolTipFinisher("GTNH Coremod");
+        if (!Keyboard.isKeyDown(Keyboard.KEY_LSHIFT)) {
+            return tt.getInformation();
+        } else {
+            return tt.getStructureInformation();
+        }
```

Comment:
Would be nice if builder had a method for optional/hidden/alternative info

Reply:
What do you mean? because it seems many multi do the same as i did.